ksqldb-engine/src/main/java/io/confluent/ksql/logicalplanner/nodes/SourceNode.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright 2022 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"); you may not use
+ * this file except in compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.logicalplanner.nodes;
+
+import com.google.common.collect.ImmutableList;
+import io.confluent.ksql.name.SourceName;
+import io.confluent.ksql.schema.ksql.LogicalColumn;
+import io.confluent.ksql.schema.ksql.LogicalSchema;
+import java.util.Objects;
+
+public abstract class SourceNode<SourceT extends SourceNode<?>> implements Node<SourceT> {
+  final SourceName sourceName;
+  final LogicalSchema simpleSchema; // schema without system columns
+  final ImmutableList<LogicalColumn> outputSchema; // to-do: should we include system columns?
```

Comment:
What are some examples of system columns?  Would those include headers and/or timestamps?  If so, I imagine the answer is "yes".

Reply:
Yes, it's `ROWTIME`, `ROWOFFSET`, `ROWPARTITION` etc.

It's not clear for me atm, how the old code exactly treat them -- they are actually added at some point in the old code -- still need to figure out / understand the details.

My gut feeling is also that the answer might be "yes".